File: java/src/main/java/no/bouvet/workshop/solid/delegation/Entry.java
package no.bouvet.workshop.solid.delegation;

public class Entry {

    private final String fromIp;
    private final String toIp;
    private final String locationFrom;
    private final String locationTo;
    private final String countryCode;
    private final String country;

    public Entry(
            String fromIp,
            String toIp,
            String locationFrom,
            String locationTo,
            String countryCode,
            String country
    ) {
        this.fromIp = fromIp;
        this.toIp = toIp;
        this.locationFrom = locationFrom;
        this.locationTo = locationTo;
        this.countryCode = countryCode;
        this.country = country;
    }

    public String getFromIp() {
        return fromIp;
    }

    public String getToIp() {
        return toIp;
    }

    public String getLocationFrom() {
        return locationFrom;
    }

    public String getLocationTo() {
        return locationTo;
    }

    public String getCountryCode() {
        return countryCode;
    }

    public String getCountry() {
        return country;
    }
}

File: dotNet/delegation/delegation/delegation/Entry.cs
namespace delegation.delegation
{
    public class Entry
    {
        private readonly string fromIp;
        private readonly string toIp;
        private readonly string locationFrom;
        private readonly string locationTo;
        private readonly string countryCode;
        private readonly string country;

        public Entry(
                string fromIp,
                string toIp,
                string locationFrom,
                string locationTo,
                string countryCode,
                string country
        )
        {
            this.fromIp = fromIp;
            this.toIp = toIp;
            this.locationFrom = locationFrom;
            this.locationTo = locationTo;
            this.countryCode = countryCode;
            this.country = country;
        }

        public string getFromIp()
        {
            return fromIp;
        }

        public string getToIp()
        {
            return toIp;
        }

        public string getLocationFrom()
        {
            return locationFrom;
        }

        public string getLocationTo()
        {
            return locationTo;
        }

        public string getCountryCode()
        {
            return countryCode;
        }

        public string getCountry()
        {
            return country;
        }
    }

}
